src/DynamoCore/Migration/Migration.cs

```diff
@@ -321,22 +321,16 @@ orderby result.From
             var migrationData = new NodeMigrationData(elNode.OwnerDocument);
             migrationData.AppendNode(elNode as XmlElement);
 
-            while (workspaceVersion != null && workspaceVersion < currentVersion)
+            foreach(var migration in migrations.Where(m=>m.From >= workspaceVersion))
```

Comment:
seems to make sense - does it break any existing migration tests here or for revit?